hibernate-core/src/main/java/org/hibernate/mapping/Column.java

```diff
@@ -425,6 +435,11 @@ public String getCustomRead() {
 		return customRead;
 	}
 
+	public void setResolvedCustomRead(String customRead) {
+		assert customRead == null || ! customRead.trim().equals( "" );
```

Comment:
As https://github.com/hibernate/hibernate-orm/pull/3491 suggested, we can replace
```
!customRead.trim().equals( "" );
```
with
```
customRead.trim().length() > 0;
```
As reported in https://medium.com/javarevisited/micro-optimizations-in-java-string-equals-22be19fd8416, the performance boot is nontrivial. @Sanne even created some util method for this.

Reply:
Not super important, but it's worth pointing out that if we had JDK11, you could use:
 - https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#isBlank()

Of course I do agree that it's not worth dropping 8 compatibility just for this, but it's a good example of the small many things, which will eventually add up into a more efficient library.

I also think it's a good example because it's simply not worth it to have two(more) copies of this class if we wanted to use multi-release jars: I suppose we all agree that it would be insane to maintain two copies of the same sources just for such a little optimisation.

Reply:
Ahem. Folks, it should actually use `StringHelper.isEmptyOrWhitespace()`.

When we use util methods for stuff like this it makes it easy to optimize them later.

Reply:
As for performance... its an assert... ;)

Reply:
Well, actually @gavinking - the check is not whether the text is empty or a whitespace... the check is that the text is either null or non-empty ("" should fail the check here)

Reply:
Nm, you probably just mean the second condition

Reply:
ah, haha right, sorry :)